File: component/src/main/java/com/galaxybruce/component/net/exception/AppNetException.java
package com.galaxybruce.component.net.exception;


/**
 * description：异常的统一处理
 * <p>
 * author： DS.Hu
 * <p>
 * time： 2018/7/23 18:58
 * <p>
 */

public class AppNetException extends Exception {
    private String code;

    public AppNetException(String code, String message) {
        super(message);
        this.code = code;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }
}

File: component/src/main/java/com/galaxybruce/component/net/exception/AppNetException.kt
package com.galaxybruce.component.net.exception

class AppNetException : Exception {
    var code: String

    constructor(code: String, message: String?) : super(message) {
        this.code = code
    }

    constructor(code: Int, message: String?) : super(message) {
        this.code = code.toString()
    }

    constructor(netError: AppNetError, e: Throwable? = null) : super("${netError.msg}${if(e?.message.isNullOrEmpty()) "" else "[${e?.message}]"}") {
        this.code = netError.code
    }
}